astropy/table/column.py

```diff
@@ -343,6 +343,79 @@ class ColumnInfo(BaseColumnInfo):
     """
     attrs_from_parent = BaseColumnInfo.attr_names
     _supports_indexing = True
+    # For structured columns, data is used to store a dict of columns.
+    # Store entries in that dict as name.key instead of name.data.key.
+    _represent_as_dict_primary_data = 'data'
+
+    def _represent_as_dict(self):
+        result = super()._represent_as_dict()
+        if names := self._parent.dtype.names:
```

Comment:
This is a test that it's structured? then https://numpy.org/doc/stable/user/basics.rec.html#manipulating-and-displaying-structured-datatypes recommends a check with `None`.
"The recommended way to test if a dtype is structured is with if dt.names is not None rather than if dt.names, to account for dtypes with 0 fields."

```suggestion
        if (names := self._parent.dtype.names) is not None:
```

Reply:
Done